src/main/java/subway/dao/H2SectionDao.java

```diff
@@ -0,0 +1,115 @@
+package subway.dao;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import org.springframework.stereotype.Repository;
+import subway.entity.SectionEntity;
+import subway.exception.ApiNoSuchResourceException;
+import subway.exception.NoSuchLineException;
+
+import javax.sql.DataSource;
+import java.sql.PreparedStatement;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Repository
+public class H2SectionDao implements SectionDao {
+
+    private final JdbcTemplate jdbcTemplate;
+    private final SimpleJdbcInsert insertAction;
+
+    private final RowMapper<SectionEntity> rowMapper = (rs, rowNum) ->
+            new SectionEntity(
+                    rs.getLong("id"),
+                    rs.getLong("line_id"),
+                    rs.getLong("up_station_id"),
+                    rs.getLong("down_station_id"),
+                    rs.getInt("distance")
+            );
+
+    public H2SectionDao(final JdbcTemplate jdbcTemplate, final DataSource dataSource) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.insertAction = new SimpleJdbcInsert(dataSource)
+                .withTableName("section")
+                .usingGeneratedKeyColumns("id");
+    }
+
+    @Override
+    public SectionEntity insert(final SectionEntity sectionEntity) {
+        Map<String, Object> params = new HashMap<>();
+        params.put("line_id", sectionEntity.getLineId());
+        params.put("up_station_id", sectionEntity.getUpStationId());
+        params.put("down_station_id", sectionEntity.getDownStationId());
+        params.put("distance", sectionEntity.getDistance());
+
+        Long sectionId = insertAction.executeAndReturnKey(params).longValue();
+        return new SectionEntity(sectionId, sectionEntity.getLineId(), sectionEntity.getUpStationId(),
+                sectionEntity.getDownStationId(),
+                sectionEntity.getDistance());
+    }
+
+    @Override
+    public void insertAll(final List<SectionEntity> sectionEntities) {
+        String sql = "INSERT INTO section (line_id, up_station_id, down_station_id, distance) "
+                + "VALUES (?,?,?,?)";
+
+        jdbcTemplate.batchUpdate(sql, sectionEntities, sectionEntities.size(),
+                (PreparedStatement ps, SectionEntity sectionEntity) -> {
+                    ps.setLong(1, sectionEntity.getLineId());
+                    ps.setLong(2, sectionEntity.getUpStationId());
+                    ps.setLong(3, sectionEntity.getDownStationId());
+                    ps.setInt(4, sectionEntity.getDistance());
+                });
```

Comment:
```suggestion
final BeanPropertySqlParameterSource[] parameterSources = sectionEntities.stream()
            .map(BeanPropertySqlParameterSource::new)
            .toArray(BeanPropertySqlParameterSource[]::new);
insertAction.executeBatch(parameterSources);
```

SimpleJdbcInsert 의존성을 가지고 있으니 이를 사용해서 batchInsert를 해보는건 어때?